### DISCUSSION TITLE
Removing item from collection list, using Livewire

### DISCUSSION THREAD

**Discussion by alex-r-redfern** [spatie/laravel-medialibrary#2957] [Q&A]
Hi there,

I'm following the example [here](https://spatie.be/docs/laravel-medialibrary/v9/handling-uploads-with-media-library-pro/handling-uploads-with-livewire) for a media collection which is working well, and I have implemented the custom list and item views.
However, I'm having issues on how to remove an item from the collection and have the displayed collection updated.

I have taken two approaches:
1) On the item have a button with 
```
wire:click="$emitTo('user.documents', 'deleteDocument', {{ json_encode($mediaItem['uuid']) }})"
```
which is passed along to a listener in the component that is as follows:

```
public function deleteDocument($uuid)
    {
        $this->user->syncFromMediaLibraryRequest($this->documents)
            ->toMediaCollection('documents');
        $this->emit('refreshComponent');

    } 
```
This does indeed delete the media file, but the frontend shows no change until I perform a page refresh

2) Having the button call `wire:click="remove('{{ $mediaItem['uuid'] }}')` which pipes to the parent `LivewireMediaLibraryComponent`'s `remove` method, that dumps the media item from the `$this->media` collection, but this doesn't persist.

I'm sure I'm missing something obvious, but does anyone have any advice on this?

Thanks very much,

Alex

**Comment by SergeyJar** [comment]
managed to solve the problem?

**Reply by alex-r-redfern** [reply]
It boiled down to us using a custom template that was missing some of the livewire bindings that the native templates did.
If I recall correctly, it was that the individual media item in the collection list element was missing a `wire:key="item-{{ $mediaItem->uuid }}"`
Hope this helps
